src/btreemap/iter.rs

```diff
@@ -71,19 +73,10 @@ where
             range,
         }
     }
-}
 
-impl<K, V, M> Iterator for Iter<'_, K, V, M>
-where
-    K: Storable + Ord + Clone,
-    V: Storable,
-    M: Memory,
-{
-    type Item = (K, V);
-
-    fn next(&mut self) -> Option<Self::Item> {
-        match self.cursors.pop() {
-            Some(Cursor::Address(address)) => {
+    fn update_cursors_for_iteration(&mut self, cursor: Cursor<K>) {
```

Comment:
This part is extracted from previous implementation of `fn next()` so it can be reused for both `next()` and `next_without_loading_value()`.